Quadriga/src/main/java/edu/asu/spring/quadriga/web/config/SimpleSignInAdapter.java

```diff
@@ -0,0 +1,127 @@
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.exceptions.UsernameExistsException;
+import edu.asu.spring.quadriga.service.IUserManager;
+import edu.asu.spring.quadriga.service.QuadrigaUserDetails;
+import edu.asu.spring.quadriga.web.config.social.SocialSignInStatus;
+import edu.asu.spring.quadriga.web.login.QuadrigaGrantedAuthority;
+import edu.asu.spring.quadriga.web.manageusers.beans.AccountRequest;
+
+/**
+ * 
+ * This class authorizes (assigns appropriate roles) users authenticated by
+ * Github.
+ * 
+ * @author Chiraag Subramanian
+ *
+ */
+
+public final class SimpleSignInAdapter implements SignInAdapter {
+
+    private final Logger logger = LoggerFactory.getLogger(SignInAdapter.class);
+
+    private IUserManager userManager;
+    private IUserHelper userHelper;
+
+    public SimpleSignInAdapter(IUserManager userManager, IUserHelper userHelper) {
+        this.userManager = userManager;
+        this.userHelper = userHelper;
+    }
+
+    /**
+     * This method checks if Quadriga contains details of the user authenticated
+     * by Github. If the user details are present (which indicates the user is
+     * an authorized user), appropriate roles are assigned, Else if the user
+     * details are not present (which indicates the user is an unregistered /
+     * not approved user, restricted access is assigned)
+     */
+
+    @Override
+    public String signIn(String userId, Connection<?> connection, NativeWebRequest request) {
+        List<GrantedAuthority> authorities = new ArrayList<>();
+        SocialSignInStatus messageType = null;
+        IUser user = (IUser) userManager.findUserByProviderUserId(connection.getKey().getProviderUserId(),
+                connection.getKey().getProviderId());
+        // if user details is not present in the database, create a database
+        // record for the user.
+        if (user == null) {
+            user = userHelper.createUser(connection);
+            try {
+                
+                if(userManager.checkUserRequestExists(user.getUserName())){
+                    messageType = SocialSignInStatus.REGISTRATION_SUCCESS;
+                }
+                else{
+                    AccountRequest accountRequest = new AccountRequest();
+                    accountRequest.setUsername(user.getUserName());
+                    accountRequest.setName(user.getName());
+                    accountRequest.setEmail(user.getEmail());
+                    accountRequest.setSocialSignIn(true);
+                    accountRequest.setProvider(user.getProvider());
+                    accountRequest.setUserIdOfProvider(user.getUserIdOfProvider());
+            
+                    if(userManager.addNewUser(accountRequest) ){
+                        messageType = SocialSignInStatus.REGISTRATION_SUCCESS;
+                    }
+                }
+            } catch (QuadrigaStorageException | UsernameExistsException e) {
```

Comment:
these are two different situations and should be handled accordingly. If there is a storage exception, something is wrong with the backend (requires appropriate log message). If the username already exists, then we have a weird case were apparently the same user tried to register twice at the same time or something, this also needs an appropriate error message.